client/src/commands.ts

```diff
@@ -116,6 +116,42 @@ function getTemplateTcb(
   };
 }
 
+/**
+ * Command goToComponentWithTemplateFile finds components which reference an external template in
+ * their `templateUrl`s.
+ *
+ * @param ngClient LSP client for the active session
+ */
+function goToComponentWithTemplateFile(ngClient: AngularLanguageClient): Command {
+  return {
+    id: 'angular.goToComponentWithTemplateFile',
+    isTextEditorCommand: true,
+    async execute(textEditor: vscode.TextEditor) {
+      const componentLocations = await ngClient.getComponentsForOpenExternalTemplate(textEditor);
+      if (componentLocations === undefined) {
+        return;
+      }
+
+      const locations: vscode.Location[] =
+          componentLocations.map(location => new vscode.Location(location.uri, location.range));
+      // If there is more than one component that references the template, show them all. Otherwise
+      // go to the component immediately.
+      if (locations.length > 1) {
+        vscode.commands.executeCommand(
+            'editor.action.showReferences',
+            textEditor.document.uri,
+            new vscode.Position(
+                textEditor.selection.start.line, textEditor.selection.start.character),
+            locations,
+        );
+      } else {
+        const document = await vscode.workspace.openTextDocument(locations[0].uri);
+        await vscode.window.showTextDocument(document, {selection: locations[0].range});
```

Comment:
Does this replace the current editor or show it in a separate panel?
Go to definition from `templateUrl` opens the file in a separate editor, so the behavior here should be consistent with that as well.

Reply:
Updated to use the `goToLocations` instead, which handles multiple results already rather than having this logic here. also tested to ensure this operates the same as a "go to definition" operation with respect to pinned/unpinned files.